docs/contracts/function-modifiers.rst

```diff
@@ -109,7 +109,17 @@ Multiple modifiers are applied to a function by specifying them in a
 whitespace-separated list and are evaluated in the order presented.
 
 Modifiers cannot implicitly access or change the arguments and return values of functions they modify.
-Their values can only be passed to them explicitly at the point of invocation.
+Those identifiers can, however, be used to explicitly pass them into the modifier as its arguments.
+When one of those values happens to be a complex value reference type, the modified members of the referenced
+value will of course be visible to all accessors of the referenced item, including the modified function and
+the scope it ultimately returns to.
+
+The reference itself and items that are non-reference value types follow the usual rules: The
+visibility of changes to them is limited to the lexical scope of the modifier function. You
+can think of the called function as shadowing the symbol of the same name. Note that
+as a consequence, that means changes to the value made within the
+``_;`` (i.e. the modified function) are not visible within the modifier function, even
+after the ``_;``.
```

Comment:
I don't think this part is helpful, but rather confusing here. Basically, regular reference and value semantics applies.